edgraph/server.go

```diff
@@ -1570,6 +1648,73 @@ func parseRequest(qc *queryContext) error {
 	return validateQuery(qc.dqlRes.Query)
 }
 
+// addQueryIfUnique adds dummy queries in the request for checking whether predicate is unique in the db
```

Comment:
This may works fine when no concurrent mutations or unique value already exist into db. But think about this, there may have two concurrent mutations which insert the same value which not exist yet into db, and below query "__dgraph_uniquecheck_%v__" can't find duplicate value, eventually there would be two same value inserted into db.

This may called MVCC duplicate key problem.

As i think, there may be two resolutions:
1. We first write duplicated values into db, but not commit, and then find way to read uncommited values, then find the unique value conflict and return unique key conflict error.
2. We set up an unique key buffer predicate into db, write and commit duplicate values into buffer predicate before execute mutations, and do unique check before commit and delete unique value from buffer after commit.

Reply:
Very pleasure to hear that. Infect, we are using @upsert to achieve the unique constrain, it works well except a lot of "transaction abort" error, and we have to keep retrying until success.

Thank you!